src/python/WMComponent/TaskArchiver/CleanCouchPoller.py

```diff
@@ -362,12 +372,23 @@ def cleanAlreadyArchivedWorkflows(self):
             requestNames = [x['key'] for x in results]
             logging.info("There are %s workflows to check for archived status", len(requestNames))
 
-            workflowDict = self.centralRequestDBReader.getStatusAndTypeByRequest(requestNames)
-
-            for request, value in viewitems(workflowDict):
-                if value[0].endswith("-archived"):
+            if not self.useReqMgrForCompletionCheck:
+                # NOTE: For T0 we build the list of workflows only for *-archived
+                #       statuses since config.TaskArchiver.cleanCouchDelayHours
+                workflowList = []
+                startTime = int(time.time()) - int(self.cleanCouchDelayHours * 60)
+                for status in ['normal-archived', 'rejected-archived', 'aborted-archived']:
```

Comment:
I am pretty sure T0 workflows don't have these statuses. Please check withe German or Jhonatan what are the very final/archived status for T0 workflows.

Actually, do we need to change the logic for fetching workflows from local wmstats in this method? I think we only need to change the comparison against (-archived) status. Could you please investigate it?

Reply:
> I am pretty sure T0 workflows don't have these statuses. 
You are partially correct I think. I crosschecked with @germanfgv and I also did the following short experiment:
```
In [19]: from WMCore.Services.RequestDB.RequestDBReader import RequestDBReader
In [20]: centralRequestDBURL = 'https://cmsweb-testbed.cern.ch/couchdb/t0_request'
In [21]: RequestCouchApp = 'T0Request'
In [22]: centralRequestDBReader = RequestDBReader(centralRequestDBURL, couchapp=RequestCouchApp)

In [40]: days = 10*365; startTime = int(time.time()) - days*24*60*60

In [41]: len(centralRequestDBReader.getRequestByStatusAndStartTime("aborted-archived", startTime=startTime, detail=False))
Out[41]: 0
In [42]: len(centralRequestDBReader.getRequestByStatusAndStartTime("rejected-archived", startTime=startTime, detail=False))
Out[42]: 0
In [43]: len(centralRequestDBReader.getRequestByStatusAndStartTime("normal-archived", startTime=startTime, detail=False))
Out[43]: 13677
```

So it seems for T0 everything goes to `normal-archived` at the end.

As of:
> Actually, do we need to change the logic for fetching workflows from local wmstats in this method

WE are not changing local with central CouchDB (or vise versa) with this new call.  We simply change the method we  are querying for archived workflows for T0, so that we do query for archived workflows only for the given period. And actually when I looked deeper into the code the previous call was not less expensive.

Reply:
Ok, actually, I do see what you mean ..... by changing from local to central couch. I need to think about that. I'll reply in a while.